pyeffects/Future.py

```diff
@@ -13,7 +13,7 @@
 from functools import reduce
 import threading
 
-A = TypeVar('A', covariant=True)
+A = TypeVar('A')
```

Comment:
Mypy fails if A is covariant, but I should be it, isn't it?

Reply:
I think it should covariant be b/c if `Dog` is a type of `Animal`, then we'd want a `Future[Dog]` to be a proper subtype of `Future[Animal]`.

The same error is occurring via `mypy` for the other classes too.

This ties back to the idea that all of these classes extend `Monad` which in turn is a type of `Generic[A]`.

I have a fix, but it involves ignoring two lines.

I'll merge this in and submit a separate PR to fix the linter issues.

The problem with covariant types is that they can't be used as input parameters to functions, b/c those types need to be either contravariant types.

Reply:
Okay, got it. Thank you.